### DISCUSSION TITLE
Tested on smaller JSON?

### DISCUSSION THREAD

**Discussion by tomByrer** [simdjson/simdjson#1407] [General]
One of my applications for this would have very random length JSON objects.  Have you tested this how it fares with smaller JSON vs other libs?

This project seems very exciting.  I used to program DSP, so I know how fast SIMD can be, though there is a small translation cost.

**Comment by jkeiser** [comment]
Yep. I don't know where the results are, but when @lemire tested it, simdjson was the fastest implementation even for very  small files (though its margin was much smaller the smaller the files got). I used to worry about this too; but given those results, I don't think there's actually much in the way of overhead.

**Comment by lemire** [comment]
See one of the figures in the readme which addresses this concern. We initially focused on "sizeable files", but later found out that we scaled well to larger or smaller files.
